### ISSUE TITLE
Install of yay 10.2.2 fails when using yay 10.2.1

### ISSUE THREAD

**Issue by marcelroed** [Jguer/yay#1506]
### Affected Version

```sh
$ yay -V
yay v10.2.1 - libalpm v12.0.1
```

### Describe the bug

When installing 10.2.2-1 using 10.2.1 I get the following error.

### Reproduction Steps
1. Have yay 10.2.1
2. Run `yay -S yay`
3. Install fails

This seems particular to my system, but I cannot for the life of me understand why it happens.

### Expected behavior

Successful installation

### Output

After running `yay -S yay`
```sh
...
==> Validating source files with sha256sums...
    yay-10.2.2.tar.gz ... Passed
==> Making package: yay 10.2.2-1 (Fri 14 May 2021 09:53:23 PM CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found yay-10.2.2.tar.gz
==> Validating source files with sha256sums...
    yay-10.2.2.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting yay-10.2.2.tar.gz with bsdtar
==> Sources are ready.
==> Making package: yay 10.2.2-1 (Fri 14 May 2021 09:53:31 PM CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
go build -trimpath -mod=readonly -modcacherw -tags six -ldflags '-X "main.yayVersion=10.2.2" -X "main.localePath=/usr/share/locale/" -linkmode=external' -buildmode=pie -o yay
go: downloading github.com/Jguer/go-alpm/v2 v2.0.5
go: downloading github.com/leonelquinteros/gotext v1.4.0
go: downloading github.com/Morganamilo/go-srcinfo v1.0.0
go: downloading github.com/mikkeloscar/aur v0.0.0-20200113170522-1cb4e2949656
go: downloading github.com/Morganamilo/go-pacmanconf v0.0.0-20210502114700-cff030e927a5
go: downloading github.com/pkg/errors v0.9.1
go: downloading golang.org/x/term v0.0.0-20210503060354-a79de5458b56
go: downloading github.com/stretchr/testify v1.7.0
go: downloading golang.org/x/sys v0.0.0-20210507161434-a76c4d0a0096
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
# github.com/Jguer/go-alpm/v2
../gopath/pkg/mod/github.com/!jguer/go-alpm/v2@v2.0.5/handle_six.go:32:10: could not determine kind of name for C.alpm_option_add_architecture
../gopath/pkg/mod/github.com/!jguer/go-alpm/v2@v2.0.5/handle_six.go:20:10: could not determine kind of name for C.alpm_option_get_architectures
../gopath/pkg/mod/github.com/!jguer/go-alpm/v2@v2.0.5/handle_six.go:38:10: could not determine kind of name for C.alpm_option_remove_architecture
../gopath/pkg/mod/github.com/!jguer/go-alpm/v2@v2.0.5/handle_six.go:26:10: could not determine kind of name for C.alpm_option_set_architectures
make: *** [Makefile:112: yay] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
error making: yay
```
Configuration
```sh
$ yay -Pg
{
	"aururl": "https://aur.archlinux.org",
	"buildDir": "/home/marcelroed/.cache/yay",
	"absdir": "/home/marcelroed/.cache/yay/abs",
	"editor": "",
	"editorflags": "",
	"makepkgbin": "makepkg",
	"makepkgconf": "",
	"pacmanbin": "pacman",
	"pacmanconf": "/etc/pacman.conf",
	"redownload": "no",
	"rebuild": "no",
	"answerclean": "",
	"answerdiff": "",
	"answeredit": "",
	"answerupgrade": "",
	"gitbin": "git",
	"gpgbin": "gpg",
	"gpgflags": "",
	"mflags": "",
	"sortby": "votes",
	"searchby": "name-desc",
	"gitflags": "",
	"removemake": "ask",
	"sudobin": "sudo",
	"sudoflags": "",
	"requestsplitn": 150,
	"sortmode": 0,
	"completionrefreshtime": 7,
	"sudoloop": false,
	"timeupdate": false,
	"devel": false,
	"cleanAfter": false,
	"provides": true,
	"pgpfetch": true,
	"upgrademenu": true,
	"cleanmenu": true,
	"diffmenu": true,
	"editmenu": false,
	"combinedupgrade": false,
	"useask": false,
	"batchinstall": false
}
```
Go version
```sh
$ go version
go version go1.16.4 linux/amd64
```

**Comment by marcelroed** [comment]
Fixed by downgrading `pacman 6.0.0alpha1-1` to `5.2.2-3`

**Comment by Jguer** [comment]
> Fixed by downgrading `pacman 6.0.0alpha1-1` to `5.2.2-3`

If you want to use pacman 6 capabilities you can install `pacman-git` (you'll need to recompile yay)
